tests/test_other.py

```diff
@@ -2260,17 +2260,32 @@ def test_debuginfo(self):
           else:
             self.assertIn('strip-debug', opts)
 
-  def test_scons(self): # also incidentally tests c++11 integration in llvm 3.1
+  def test_scons(self):
+    # this test copies the site_scons directory alongside the test
     scons_path = Building.which('scons')
     if not scons_path:
       self.skipTest('Skipping other.test_scons: The tool "scons" was not found in PATH!')
-    try_delete(os.path.join(self.get_dir(), 'test'))
-    shutil.copytree(path_from_root('tests', 'scons'), os.path.join(self.get_dir(), 'test'))
-    shutil.copytree(path_from_root('tools', 'scons', 'site_scons'), os.path.join(self.get_dir(), 'test', 'site_scons'))
-    os.chdir(os.path.join(self.get_dir(), 'test'))
-    run_process(['scons'])
-    output = run_js('scons_integration.js', assert_returncode=5)
-    assert 'If you see this - the world is all right!' in output
+    shutil.copytree(path_from_root('tests', 'scons'), 'test')
+    shutil.copytree(path_from_root('tools', 'scons', 'site_scons'), os.path.join('test', 'site_scons'))
+    with chdir('test'):
+      with env_modify({'EMSCRIPTEN_ROOT': path_from_root()}):
```

Comment:
It'd be neat if there were some way to `with` both of these at once...
Oh hey SO says you can: https://stackoverflow.com/questions/3024925/python-create-a-with-block-on-several-context-managers
Needs python 2.7, but that should be fine?
`with chdir('test'), env_modify({...`